src/app/services/templates.service.ts

```diff
@@ -0,0 +1,169 @@
+
+    loadTemplates() {
+        if (this.loading) {
+            return;
+        }
+        this.loading = true;
+
+        this.backend.getPluginEndpoints().subscribe(pluginEndpoints => {
+            var observables: Observable<TemplateDescription>[] = [];
+            pluginEndpoints.items.map(pluginEndpoint => {
+                if (pluginEndpoint.type === "PluginRunner") {
+                    observables.push(this.http.get<{ templates: TemplateDescription[] }>(`${pluginEndpoint.url}/templates`).pipe(
+                        mergeMap(templateResponse => from(templateResponse.templates)),
+                        catchError(err => {
+                            console.log(err);
+                            return [];
+                        })
+                    ))
+                }
+            });
+
+            observables.push(of({
+                name: 'All Plugins',
+                description: 'Display All Loaded Plugins',
+                identifier: 'all-plugins',
+                categories: [
+                    {
+                        name: 'All Plugins',
+                        description: 'Display All Loaded Plugins',
+                        identifier: 'all-plugins',
+                        pluginFilter: true,
+                    }
+                ]
+            }))
+
+            from(observables).pipe(
+                mergeAll(),
+                toArray(),
+            ).subscribe(templates => {
+                templates.sort((a, b) => a.name.localeCompare(b.name));
+                this.templatesSubject.next(templates);
+                this.loading = false;
+            });
+        })
+    }
+}
+
+function isInstanceOfPluginFilterOr(pluginFilter: PluginFilterExpr): pluginFilter is PluginFilterOr {
+    return pluginFilter != null && (pluginFilter as PluginFilterOr).or !== undefined;
+}
+
+function isInstanceOfPluginFilterAnd(pluginFilter: PluginFilterExpr): pluginFilter is PluginFilterAnd {
+    return pluginFilter != null && (pluginFilter as PluginFilterAnd).and !== undefined;
+}
+
+function isInstanceOfPluginFilterNot(pluginFilter: PluginFilterExpr): pluginFilter is PluginFilterNot {
+    return pluginFilter != null && (pluginFilter as PluginFilterNot).not !== undefined;
+}
+
+function isInstanceOfString(pluginFilter: PluginFilterExpr): pluginFilter is string {
+    return pluginFilter != null && typeof pluginFilter === 'string';
+}
+
+function isInstanceOfBoolean(pluginFilter: PluginFilterExpr): pluginFilter is boolean {
+    return pluginFilter != null && typeof pluginFilter === 'boolean';
+}
+
+export function pluginMatchesFilter(pluginDesc: PluginDescription, pluginFilter: PluginFilterExpr): boolean {
+    if (isInstanceOfPluginFilterOr(pluginFilter)) {
+        return pluginFilter.or.reduce<boolean>(
+            (res, nestedPluginFilter) => res || pluginMatchesFilter(pluginDesc, nestedPluginFilter),
+            false,
+        );
+    } else if (isInstanceOfPluginFilterAnd(pluginFilter)) {
```

Comment:
Because all `if` branches immediately return there is no need for an `else if` chain. Just write all if conditions.